development/gcp/cloud-functions/rotateserviceaccount/rotateserviceaccount.go

```diff
@@ -0,0 +1,126 @@
+package rotateserviceaccount
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/kyma-project/test-infra/development/gcp/pkg/cloudfunctions"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/pubsub"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/secretmanager"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/secretversionsmanager"
+	"google.golang.org/api/iam/v1"
+)
+
+var (
+	projectID                   string
+	secretManagerService        *secretmanager.Service
+	secretVersionManagerService *secretversionsmanager.Service
+	serviceAccountService       *iam.Service
+)
+
+// ServiceAccountJSON stores Service Account athentication data
+type ServiceAccountJSON struct {
+	Type             string `json:"type"`
+	ProjectID        string `json:"project_id"`
+	PrivatekayID     string `json:"private_key_id"`
+	PrivateKey       string `json:"private_key"`
+	ClientEmail      string `json:"client_email"`
+	ClientID         string `json:"client_id"`
+	AuthURL          string `json:"auth_uri"`
+	TokenURI         string `json:"token_uri"`
+	AuthProviderCert string `json:"auth_provider_x509_cert_url"`
+	ClientCert       string `json:"client_x509_cert_url"`
+}
+
+func init() {
+	var err error
+	ctx := context.Background()
+
+	projectID = os.Getenv("GCP_PROJECT_ID")
```

Comment:
Are you sure this variable will *always* be set? `os.Getenv` returns `""` if variable is not set. I don't see that you are checking this variable in the code where you use it. Will the behavior when this variable is not set be accepted?

Reply:
This varaible is unavailble in Go 1.16, I'll change that to use project ID from metadata service